pkg/route/api/validation/validation_test.go

```diff
@@ -987,6 +1011,166 @@ func TestExtendedValidateRoute(t *testing.T) {
+						Certificate:              testCertificate,
+						Key:                      testPrivateKey,
+						CACertificate:            testCACertificate,
+						DestinationCACertificate: testCACertificate,
+					},
+				},
+			},
+			expectedErrors: 0,
+		},
+		{
+			name: "Expired cert invalid key",
+			route: &api.Route{
+				Spec: api.RouteSpec{
+					Host: "www.example.com",
+					TLS: &api.TLSConfig{
+						Termination: api.TLSTerminationEdge,
+						Certificate: testExpiredCertNoKey,
+					},
+				},
+			},
+			expectedErrors: 1,
+		},
+		{
+			name: "Expired cert mismatched key",
+			route: &api.Route{
+				Spec: api.RouteSpec{
+					Host: "www.example.com",
+					TLS: &api.TLSConfig{
+						Termination: api.TLSTerminationEdge,
+						Certificate: testExpiredCertNoKey,
+						Key:         testPrivateKey,
+					},
+				},
+			},
+			expectedErrors: 1,
+		},
+		{
+			name: "Expired cert invalid key reencrypt",
+			route: &api.Route{
+				Spec: api.RouteSpec{
+					Host: "www.example.com",
+					TLS: &api.TLSConfig{
+						Termination:              api.TLSTerminationReencrypt,
+						Certificate:              testExpiredCertNoKey,
+						DestinationCACertificate: testExpiredCertNoKey,
+					},
+				},
+			},
+			expectedErrors: 1,
+		},
+		{
+			name: "Expired cert with a different route host",
+			route: &api.Route{
+				Spec: api.RouteSpec{
+					Host: "think.different.test",
+					TLS: &api.TLSConfig{
+						Termination:   api.TLSTerminationEdge,
+						Certificate:   testCertificate,
+						Key:           testPrivateKey,
+						CACertificate: testCACertificate,
+					},
+				},
+			},
+			expectedErrors: 0,
+		},
+		{
+			name: "self-signed cert",
+			route: &api.Route{
+				Spec: api.RouteSpec{
+					Host: "self.signer.test",
+					TLS: &api.TLSConfig{
+						Termination: api.TLSTerminationReencrypt,
+						Certificate: testSelfSignedCert,
+						Key:         testSelfSignedKey,
+
+						DestinationCACertificate: testSelfSignedCert,
+					},
+				},
+			},
+			expectedErrors: 0,
+		},
+		{
+			name: "self-signed cert with different host",
+			route: &api.Route{
+				Spec: api.RouteSpec{
+					Host: "different.co.us",
+					TLS: &api.TLSConfig{
+						Termination: api.TLSTerminationEdge,
+						Certificate: testSelfSignedCert,
+						Key:         testSelfSignedKey,
+					},
+				},
+			},
+			expectedErrors: 0,
+		},
+		{
+			name: "future validity date 2038 cert",
+			route: &api.Route{
+				Spec: api.RouteSpec{
+					Host: "self.signer.test",
+					TLS: &api.TLSConfig{
+						Termination: api.TLSTerminationEdge,
+						Certificate: testValidInFutureCert,
+						Key:         testValidInFutureKey,
+					},
+				},
+			},
+			expectedErrors: 0,
+		},
+		{
+			name: "future validity date 2038 cert with different host",
+			route: &api.Route{
+				Spec: api.RouteSpec{
+					Host: "www.in.the.future.test",
+					TLS: &api.TLSConfig{
+						Termination: api.TLSTerminationEdge,
+						Certificate: testValidInFutureCert,
```

Comment:
it looks like this is only passing because you omit a CA, so no validation is done at all. need to have a test with CA with not-yet-valid and expired dates to make sure CA validation passes even when the cert timestamps aren't valid.